codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/DafnyLocalServiceGenerator.java

```diff
@@ -996,14 +1103,48 @@ void generateNativeResourceWrapper(
                 "dafny"
               );
             } else {
-              clientResponse = "var native_response, native_error";
-              returnResponse =
-                "%s(*native_response)".formatted(
-                    SmithyNameResolver.getToDafnyMethodName(
+              String fromDafnyConvMethodNameForOutput =
+                SmithyNameResolver.getToDafnyMethodName(
+                  service,
+                  outputShape,
+                  ""
+                );
+              boolean deReferenceRequired = true;
+              if (outputShape.hasTrait(PositionalTrait.class)) {
+                final MemberShape postionalMemShape = outputShape
+                  .getAllMembers()
+                  .values()
+                  .stream()
+                  .findFirst()
+                  .get();
+                outputShape = model.expectShape(postionalMemShape.getTarget());
+                if (outputShape.hasTrait(ReferenceTrait.class)) {
+                  outputShape =
+                    model.expectShape(
+                      outputShape
+                        .expectTrait(ReferenceTrait.class)
+                        .getReferentId()
+                    );
+                }
+                fromDafnyConvMethodNameForOutput =
+                  outputShape.isResourceShape()
+                    ? SmithyNameResolver.getFromDafnyMethodName(
                       service,
                       outputShape,
                       ""
                     )
+                    : Constants.funcNameGenerator(
+                      postionalMemShape,
+                      "FromDafny",
+                      model
+                    );
+                deReferenceRequired = false;
+              }
+              clientResponse = "var native_response, native_error";
+              returnResponse =
+                "%s(%snative_response)".formatted(
+                    fromDafnyConvMethodNameForOutput,
+                    deReferenceRequired ? "*" : ""
```

Comment:
I prefer this style over what's done at 923.

Reply:
Yes. Remove the if else and added ternary instead of it. Now its line 922